src/core/packages/elasticsearch/server-internal/src/version_check/ensure_es_version.test.ts

```diff
@@ -298,129 +328,129 @@ describe('pollEsNodesVersion', () => {
         error: done,
       });
+    subscriptions.push(subscription);
   });
 
-  describe('marble testing', () => {
-    const getTestScheduler = () =>
-      new TestScheduler((actual, expected) => {
-        expect(actual).toEqual(expected);
-      });
-
-    const mockTestSchedulerInfoResponseOnce = (infos: NodesInfo) => {
-      // @ts-expect-error we need to return an incompatible type to use the testScheduler here
-      internalClient.nodes.info.mockReturnValueOnce([infos]);
-    };
-
-    it('starts polling immediately and then every healthCheckInterval', () => {
-      expect.assertions(1);
-
-      mockTestSchedulerInfoResponseOnce(createNodes('5.1.0', '5.2.0', '5.0.0'));
-      mockTestSchedulerInfoResponseOnce(createNodes('5.1.1', '5.2.0', '5.0.0'));
-
-      getTestScheduler().run(({ expectObservable }) => {
-        const expected = 'a 99ms (b|)';
-
-        const esNodesCompatibility$ = pollEsNodesVersion({
-          internalClient,
-          healthCheckInterval: 100,
-          ignoreVersionMismatch: false,
-          kibanaVersion: KIBANA_VERSION,
-          log: mockLogger,
-        }).pipe(take(2));
-
-        expectObservable(esNodesCompatibility$).toBe(expected, {
-          a: mapNodesVersionCompatibility(
-            createNodes('5.1.0', '5.2.0', '5.0.0'),
-            KIBANA_VERSION,
-            false
-          ),
-          b: mapNodesVersionCompatibility(
-            createNodes('5.1.1', '5.2.0', '5.0.0'),
-            KIBANA_VERSION,
-            false
-          ),
+  // rewritten from marble tests to use real timers instead of fake timers
+  describe('timing and interval behavior', () => {
+    it('starts first poll immediately and then polls every healthCheckInterval', (done) => {
+      const timestamps: number[] = [];
+      const startTime = Date.now();
+
+      nodeInfosSuccessOnce(createNodes('5.1.0', '5.2.0', '5.0.0'));
+      nodeInfosSuccessOnce(createNodes('5.1.1', '5.2.0', '5.0.0'));
+
+      const subscription = pollEsNodesVersion({
+        internalClient,
+        healthCheckInterval: 50,
+        ignoreVersionMismatch: false,
+        kibanaVersion: KIBANA_VERSION,
+        log: mockLogger,
+        healthCheckRetry: 1,
+      })
+        .pipe(take(2))
+        .subscribe({
+          next: () => {
+            timestamps.push(Date.now() - startTime);
+          },
+          complete: () => {
+            // First emission should be immediate (within 10ms)
+            expect(timestamps[0]).toBeLessThan(10);
+            // Second emission should be after healthCheckInterval (50ms +/- 20ms tolerance)
+            expect(timestamps[1]).toBeGreaterThanOrEqual(40);
+            expect(timestamps[1]).toBeLessThan(80);
+            done();
+          },
+          error: done,
         });
-      });
+      subscriptions.push(subscription);
     });
 
-    it('waits for es version check requests to complete before scheduling the next one', () => {
-      expect.assertions(2);
```

Comment:
here and in other tests, if we remove the expect.assertions and our assertions happen inside a subscription then the test can pass if there's no emissions, so keeping these is critical